### ISSUE TITLE
[gatsby-source-contentful] TypeError: getNode is not a function

### ISSUE THREAD

**Issue by coreyward** [gatsbyjs/gatsby#4906]
When a top-level Contentful resource queried for in `gatsby-node.js` (used to call `createPage`) is deleted from Contentful and `gatsby develop` is restarted, the following error is thrown:

```
success onPreBootstrap — 0.153 s
⠁ Starting to fetch data from Contentful
Fetching default locale
⠐ source and transform nodesdefault local is : en-US
⠈ source and transform nodescontentTypes fetched 17
error Plugin gatsby-source-contentful returned an error

  TypeError: getNode is not a function
  
  - gatsby-node.js:90 
    [project]/[gatsby-source-contentful]/gatsby-node.js:90:37
  
  - Array.forEach
  
  - gatsby-node.js:89 _callee$
    [project]/[gatsby-source-contentful]/gatsby-node.js:89:16
  
  - next_tick.js:112 process._tickCallback
    internal/process/next_tick.js:112:11
  

⠠ source and transform nodes
```

This corresponds with these lines of the `gatsby-source-contentful` plugin:

```js
currentSyncData.deletedEntries.map(function (e) {
  return e.sys.id;
}).forEach(function (id) {
  return deleteNode(id, getNode(id)); // This is line 90 that's throwing the error
});
```

I was able to resolve the issue by removing the `.cache` folder and restarting. I don't know what's “upstream” of this `sourceNodes` call to investigate what `boundActionCreators.getNode` is. 

It looks like this also happened to @johndewolf a few weeks ago when editing Contentful content: https://github.com/gatsbyjs/gatsby/issues/4401#issuecomment-373937002

### Environment

* Gatsby version (`npm list gatsby`): 1.9.243
* gatsby-cli version (`gatsby --version`): 1.1.50
* Node.js version: 9.9.0
* Operating System: macOS High Sierra 10.13.3

**Comment by pieh** [comment]
Hmm, `getNode` is not action creator and shouldn't be extracted from `boundActionCreators` - it should be extracted from first paramater of `sourceNodes` exported function (same as getNodes is) - do you feel like you could create PR with fix? Here's source for this - https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-contentful/src/gatsby-node.js#L32-L42

**Comment by coreyward** [comment]
Just to make sure I understand, we're looking for this, correct?

```js
exports.sourceNodes = async (
  { boundActionCreators, getNode, getNodes, hasNodeChanged, store },
  { spaceId, accessToken, host }
) => {
  const {
    createNode,
    deleteNode,
    touchNode,
    setPluginStatus,
  } = boundActionCreators
```

**Comment by coreyward** [comment]
Just looked at the history for that file and it looks like it was previously there, but Kyle swapped it out for `getNodes` when updating some unrelated code in the same function, so I'm going to assume that's correct.

**Comment by coreyward** [comment]
Fixed by #4907
